### ISSUE TITLE
Save label information for seq/token classification

### ISSUE THREAD

**Issue by nbroad1881** [huggingface/peft#1486]
### Feature request

Imagine you train a model for text classification on three labels: positive, negative, neutral. You train, load, merge, stick it in a pipeline.
During prediction, your pipeline will give scores like: {"LABEL_0": 0.3, "LABEL_1": 0.5, "LABEL_2": 0.2}, so you have go back to your training code to see which label corresponds with which.

Typically, this information is stored in config.json.  When using peft, the user must remember the number of labels (typically in `config.num_labels`) and the mapping of id2label (`config.id2label`). It would be nice if this information were stored in the adapter config as something called "model_kwargs" or something.

Desired flow when loading the model:
```python
from peft import PeftModel, PeftConfig
from transformers import AutoModelForSequenceClassification

peft_model_id = "peft-id"
config = PeftConfig.from_pretrained(peft_model_id)
inference_model = AutoModelForSequenceClassification.from_pretrained(config.base_model_name_or_path, **config.model_kwargs)


inference_model = PeftModel.from_pretrained(inference_model, peft_model_id)
```

### Motivation

Easier for the user to use seq/token classification + peft

### Your contribution

tagging @BenjaminBossan  :D

**Comment by BenjaminBossan** [comment]
Probably there could be other attributes as well that are being set on the model config after loading the model. Instead of trying to catch all of them, I wonder if we could add an option to save the model config in addition to the adapter files? For instance:

```python
# train a peft model
lora_path = ...
peft_model.save_pretrained(lora_path, save_model_config=True)
# lora_path now also contains the config.json
config_loaded = AutoConfig.from_pretrained(lora_path)
inference_model = AutoModelForSequenceClassification.from_pretrained(
    config.base_model_name_or_path, config=config_loaded
)
```

**Comment by lrsbrgrn** [comment]
Hi @nbroad1881 I'm still facing this issue/wondering this behaviour 1yr later, did you find any solution? Best, LB

**Comment by nbroad1881** [comment]
Still no changes afaik. You'll have to manually update it.

**Comment by BenjaminBossan** [comment]
Hi, we thought a bit more about this issue but couldn't really find a generic way to implement this. As a workaround, would something like the suggestion below do? First, you probably have a training script similar to this:

```python
id2label = {0: "NEGATIVE", 1: "POSITIVE"}
label2id = {"NEGATIVE": 0, "POSITIVE": 1}

base_model = AutoModelForSequenceClassification.from_pretrained(
    model_id, num_labels=2, id2label=id2label, label2id=label2id
)
peft_model = get_peft_model(base_model, peft_config)
```

and an inference script like this:

```python
base_model = AutoModelForSequenceClassification.from_pretrained(
    model_id, num_labels=2
)
peft_model = PeftModel.from_pretrained(base_model, path)
```

If, in the inference script, you would pass `id2label=id2label, label2id=label2id` in the same way as in the training script, the labels should be preserved. The mappings are part of the base model config and can be saved with `base_model.config.save_pretrained`. Or in case you save the model on Hugging Face Hub, you can save the base model config by calling `base_model.config.push_to_hub`.
